### DISCUSSION TITLE
How to render a table in a flowchart?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [mermaid-js/mermaid#5320] [Q&A] [answered]
Sometimes, I want to render a table inside a flowchart. However, even Markdown tables don't render:

```mermaid
flowchart TD
	n1["`
		test | test
		-----|-----
		test | test
		`"]
```

renders as

> ![image](https://github.com/mermaid-js/mermaid/assets/42837531/87991dd3-3631-4fe2-9df8-671925371276)

without even an error for unsupported Markdown.

**Comment by Piratepie** [comment]
After taking a look myself, I think we have to take the 3 examples given in the documentation less as examples, and more as the only functionality.

In the [live editor](https://mermaid.live/) I was only able to get **bold**, _italic_, and text wrapping to work.

To be fair, the docs do read:
> which supports text formatting options such as bold and italics, and automatically wraps text within labels.

and one could argue that tables aren't "text formatting".

Checkout [this response which confirms it](https://github.com/orgs/mermaid-js/discussions/4575#discussioncomment-6383508)

**Reply by roke-julian-lockhart** [reply, marked as answer]
@Piratepie, it appears that the documentation might be incorrect, for although 

~~~.MD
```mermaid
flowchart TD
	n1["<table><thead><tr><th>TestTL</th><th>TestTR</th></tr></thead><tbody><tr><td>TestBL</td><td>TestBR</td></tr></tbody></table>"]
```
~~~

doesn't render here:

```mermaid
flowchart TD
	n1["<table><thead><tr><th>TestTL</th><th>TestTR</th></tr></thead><tbody><tr><td>TestBL</td><td>TestBR</td></tr></tbody></table>"]
```

an HTML table does via https://marketplace.visualstudio.com/_apis/public/gallery/publishers/bierner/vsextensions/markdown-mermaid/1.22.0/vspackage and https://mermaid.live/edit#pako:eNpFjrEOgkAMhl-FdDa5uF4Ig2F0QjbrUGgRIneY44ghhHf38FCn_l_y_W0XqAcW0ND0w6tuyfmkzNGit8crQuqp6iVLfSvEYbgtZqWMvjynKsQfFhHVpqivXg087y3-aKetxX8sIu6tqKt4E-EGBzDiDHUc_lvQJglCWG0EQYfI5B4IaNfg0eSHy2xr0N5NcoDpyeQl7-juyIBuqB9lfQMRak_9:

> ![image](https://github.com/mermaid-js/mermaid/assets/42837531/609108a8-a166-450e-8983-463f5c6bfef6)

However, do note that prettified HTML renders incorrectly:

~~~.MD
```mermaid
flowchart TD
	n1["
		<table>
		<thead>
			<tr>
				<th>TestTL</th>
				<th>TestTR</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>TestBL</td>
				<td>TestBR</td>
			</tr>
		</tbody>
		</table>
	"]
```
~~~

```mermaid
flowchart TD
	n1["
		<table>
		<thead>
			<tr>
				<th>TestTL</th>
				<th>TestTR</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>TestBL</td>
				<td>TestBR</td>
			</tr>
		</tbody>
		</table>
	"]
```

renders as 

> ![image](https://github.com/mermaid-js/mermaid/assets/42837531/288974ed-e722-459e-99b2-335b6bec370b)

I've filed https://github.com/mermaid-js/mermaid/issues/5328#issue-2154332628 consequently, if of use.

**Reply by Piratepie** [reply]
Hadn't even considered using HTML instead of markdown, nice thinking 👍🏻
The documentation regarding markdown support still holds tho.

The top blank space seems to have something do to with the line breaks, but that's what the issue you've created is for.
